lib/active_model_serializers/adapter/json_api.rb

```diff
@@ -294,7 +294,7 @@ def attributes_for(serializer, fields)
 
       # {http://jsonapi.org/format/#document-resource-objects Document Resource Objects}
       def resource_object_for(serializer)
-        resource_object = serializer.cache_check(self) do
+        resource_object = serializer.fetch(self) do
```

Comment:
I'm still not a fan of fetching from the cache always needing an adapter_instance, but it's not terrible.

looks like we might want to add a fetch_multi around the collection?